src/main/python/systemds/matrix/data_gen.py

```diff
@@ -102,3 +105,70 @@ def rand(sds_context: SystemDSContext, rows: int, cols: int,
         named_input_nodes['seed'] = seed
 
     return OperationNode(sds_context, 'rand', [], named_input_nodes=named_input_nodes)
+
+
+def rev(sds_context: 'SystemDSContext', X: Matrix) -> 'OperationNode':
+    """ Reverses the rows in a matrix
+
+    :param sds_context: SystemDS context
+    :param X: Input matrix
+    :return: reversed matrix
+    """
+    X._check_matrix_op()
+    return OperationNode(sds_context, 'rev', [X])
+
+
+def order(sds_context: 'SystemDSContext', X: Matrix, by: int = 1, decreasing: bool = False, index_return: bool = False) -> 'OperationNode':
+    """ Sort by a column of the matrix X in increasing/decreasing order and returns either the index or data
+
+    :param sds_context: SystemDS context
+    :param X: Input matrix
+    :param by: Column number
+    :param decreasing: If true the matrix will be sorted in decreasing order
+    :param index_return: If true, theindex numbers will be returned
+    :return: sorted matrix
+    """
+    X._check_matrix_op()
+
+    cols = X._np_array.shape[1]
+    if by > cols:
+        raise IndexError("Index {i} is out of bounds for axis 1 with size {c}".format(i=by, c=cols))
+
+    named_input_nodes = {'target': X, 'by': by, 'decreasing': str(decreasing).upper(), 'index.return': str(index_return).upper()}
+
+    return OperationNode(sds_context, 'order', [], named_input_nodes=named_input_nodes)
+
+
+def t(sds_context: 'SystemDSContext', X: Matrix) -> 'OperationNode':
+    """ Transposes the input matrix
+
+    :param sds_context: SystemDS context
+    :param X: Input matrix
+    :return: transposed matrix
+    """
+    X._check_matrix_op()
+    return OperationNode(sds_context, 't', [X])
+
+
+def cholesky(sds_context: 'SystemDSContext', X: Matrix) -> 'OperationNode':
+    """ Computes the Cholesky decomposition of a symmetric, positive definite matrix
+
+    :param sds_context: SystemDS context
+    :param X: Input matrix
+    :return: Cholesky decomposition
+    """
+    X._check_matrix_op()
+
+    # check square dimension
+    if X._np_array.shape[0] != X._np_array.shape[1]:
+        raise ValueError("Last 2 dimensions of the array must be square")
+
+    # check if mat is positive definite
+    if not np.all(np.linalg.eigvals(X._np_array)>0):
+        raise ValueError("Matrix is not positive definite")
+
+    # check if mat is symmetric
+    if not np.allclose(X._np_array, X._np_array.transpose()):
+        raise ValueError("Matrix is not symmetric")
```

Comment:
I like the checks, but the complexity of these operations grow in size of X.  (except for the first check).
This makes the overall performance of the operation slow, maybe we add an optional flag saying unsafe, that skip these checks?

Reply:
Ah ok I get your point, but I'm not sure how to catch these two scenarios (positive definite, symmetric) without these check since SystemDS will throw an exception which is not easily readable after calling `compute()`.

Reply:
The system have changed alittle, so that it should be possible to catch these exceptions.
once `compute()` is called. But i don't expect to be able to use it here in this operations content. Therefore, i suggest to still just have a flag, that specifies unsafe, that simply skips these checks.

Reply:
Can you specify what you meant by using a flag unsafe? I could just put a warning into this function saying it is unsafe to use such a matrix and just let the program continue even if the user violates the conditions. Did you mean something like this?

Reply:
Something along the lines of:
```python
def cholesky(sds_context: SystemDSContext, X: Matrix, safe: boolean = false) -> OperationNode:
```

But then again + old comments it becomes:
```python
def cholesky(safe: boolean = false) -> OperationNode:
```
if you move the operations into the matrix object